recipe-server/client/actions/preference-experiment/index.js

```diff
@@ -54,13 +59,15 @@ export default class PreferenceExperimentAction extends Action {
 
       // Otherwise, enroll!
       const branch = await this.chooseBranch(branches);
+      const experimentType = isHighPopulation ? 'normandy-exp-highpop' : 'normandy-exp';
```

Comment:
Hm, but you're validating for the `experimentType` starting with `normandy-pref` in `PreferenceExperiments.start()`. Neither of these 2 strings match that.

Unless I'm missing something, this means I'd like to see 2, maybe 3 changes:

1) fix the string `startsWith` check and/or these names so that this does pass
2) write a testcase that actually passes the 'high population' thing to the telemetry thing which would have thrown and shown something was wrong here (ie ensure we test the interaction between this code and the telemetry code, which apparently right now we don't or you would have noticed this!)
3) optionally, switch this to something that isn't so stringly-typed. Like, I get that this is JS, but if we had a set of value constants defined on `PreferenceExperiments` that we used throughout then this would be a harder mistake to make/introduce.

Reply:
Also, double quotes (or, where appropriate, template strings) please. :-)

(eslint can help with this, I believe?)

Reply:
About the quotes: This code is outside of the add-on, and so doesn't use the Firefox eslint configs. It is tied more closely to the rest of our Javascript  that for historical reasons prefers single quotes. It's a weird situation, but eslint helps 😄.

Nice catch on the test-mismatch though. I'll fix that up.

Reply:
Oh, right. Does that mean that my suggestion for a test would require testing server-to-client interaction? That might be a bit much... I don't know why I missed that the file wasn't in the add-on - maybe the `client` part of the path threw me off.

Reply:
Yeah, we don't have any tools set up to test actions against the actual add-on, unfortunately. That also means that change 3 is a bit out of reach. I think it would make sense to make this API a little less sensitive, and enforce the things we actually care about (like length limits and a normandy-prefix). I'll work on that.